etcd-manager/pkg/volumes/openstack/util.go

```diff
@@ -26,20 +27,37 @@ const (
 	openstackAddress        = "addr"
 )
 
-func GetServerFixedIP(addrs map[string]interface{}, name string) (poolAddress string, err error) {
+func getAllServerFixedIPs(addrs map[string]interface{}) []string {
+	var fixedIPs []string
 	for _, address := range addrs {
 		if addresses, ok := address.([]interface{}); ok {
 			for _, addr := range addresses {
 				addrMap := addr.(map[string]interface{})
 				if addrType, ok := addrMap[openstackExternalIPType]; ok && addrType == openstackAddressFixed {
 					if fixedIP, ok := addrMap[openstackAddress]; ok {
 						if fixedIPStr, ok := fixedIP.(string); ok {
-							return fixedIPStr, nil
+							fixedIPs = append(fixedIPs, fixedIPStr)
```

Comment:
Should return at the first IP that matches the subnet, when set.

Reply:
I am not sure about that. I can move the code to [GetServerFixedIP()](https://github.com/kubernetes-sigs/etcdadm/pull/317/files#diff-73f39bca80e8581052a90e491069fa8331e144ab25f30cadd33a958aaf25e3dfR49) again, but then this would add another layer of nesting to an already heavily nested, and at least for me hard to decipher code. This would look something like this:

https://github.com/kubernetes-sigs/etcdadm/pull/317/files#r1135238691

Although moving this to a separate function which first collects all the addresses and then filters them is more inefficient, it at least makes it more clearer what the code does IMO. But probably there is a better way of doing it, which I am not aware of?